lib/dpu.rb

```diff
@@ -21,10 +22,13 @@ def determine_permanent_uri(path_or_link, start_line_number = nil, end_line_numb
       path = path_or_link.realpath
       relative_path = determine_relative_path(path)
 
+      version = find_same_content_version(path, relative_path)
+      ref = version&.ascii_only? ? version : determine_commit_id(path)
```

Comment:
Umm, if `1.0.0⚪︎` and `0.10.0` tags are exists and same content, it will call `determine_commit_id(path)` .

Following code in `find_same_content_version` method solves it, I think. :eyes:

```ruby
versions = VersionSorter.sort(stdout.each_line(chomp: true).select(&:ascii_only?))
```

Reply:
930a37dbd1d10da0e0b444c3da57c4594c1a1264

I couldn't assume the case, your suggestion would be better!